File: sentimental/caesar/caesar.py
import sys
from cs50 import get_string

print("plaintext: ", end="")

key = int(sys.argv[1])
if key < 0:
    print("Usage: Key must be a positive number")
    sys.exit(1)

while True:
    plaintxt = get_string()
    if plaintxt == 0:
        continue
    else:
        break

plain = [char for char in plaintxt]
ciphertxt = []

for char in plain:
    if char.isupper():
        ciphertxt.append(chr(((ord(char) - ord('A') + key) % 26) + ord('A')))
    elif char.islower():
        ciphertxt.append(chr(((ord(char) - ord('a') + key) % 26) + ord('a')))
    else:
        ciphertxt.append(char)

print("ciphertext: ", end="")
print("".join(ciphertxt))

File: caesar/more/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, char* argv[])
{
    int c = 0;

    if(argv[1] > 0 && argc == 2)
    {
        int inputKey = atoi(argv[1]);
        string plaintext = get_string("plaintext: ");
        printf("ciphertext: ");

        for(int i=0;i<strlen(plaintext);i++)
        {
            if((plaintext[i] >= 'a' && plaintext[i] <= 'z') || (plaintext[i] >= 'A' && plaintext[i] <= 'Z') )
            {
                if(isupper(plaintext[i]))
                {
                    c = (plaintext[i] - 65 + inputKey) % 26;
                    c += 65;
                }else if(islower(plaintext[i]))
                {
                    c = (plaintext[i] - 97 + inputKey) % 26;
                    c += 97;
                }
                printf("%c", c);
            }else
            {
                printf("%c",plaintext[i]);
            }
        }
        printf("\n");
        return 0;
    }else
        printf("Usage: ./caesar key\n");
        return 1;
}